src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,665 @@
+            }
+
+            if (tfType == TFDataType.Double)
+            {
+                return CreateTensorValueGetter<double>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Int32)
+            {
+                return CreateTensorValueGetter<DvInt4>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Int64)
+            {
+                return CreateTensorValueGetter<Int64>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Bool)
+            {
+                return CreateTensorValueGetter<bool>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.String)
+            {
+                return CreateTensorValueGetter<string>(input, type, colIndex, tfShape);
+            }
+
+            throw Host.ExceptNotSupp("Tensorflow type not supported");
+        }
+
+        private ITensorValueGetter[] GetTensorValueGetters(IRow input, int iinfo)
+        {
+            var info = _bindings.Infos[iinfo];
+            var tfInfo = _bindings.TfColInfo[iinfo];
+            var srcTensorGetters = new ITensorValueGetter[info.SrcIndices.Length];
+            for (int j = 0; j < info.SrcIndices.Length; j++)
+            {
+                int colIndex = _bindings.Infos[iinfo].SrcIndices[j];
+                srcTensorGetters[j] = CreateTensorValueGetterVec(input, tfInfo.TfTypes[j], info.SrcTypes[j], colIndex, tfInfo.TfShapes[j]);
+            }
+            return srcTensorGetters;
+        }
+
+        private unsafe T[] FetchData<T>(IntPtr data, int size)
+        {
+            var result = new T[size];
+
+            GCHandle handle = GCHandle.Alloc(result, GCHandleType.Pinned);
+            IntPtr target = handle.AddrOfPinnedObject();
+
+            Int64 sizeInBytes = size * Marshal.SizeOf((typeof(T)));
+            Buffer.MemoryCopy(data.ToPointer(), target.ToPointer(), sizeInBytes, sizeInBytes);
+            handle.Free();
+            return result;
+        }
+
+        private Delegate MakeGetter(IRow input, int iinfo)
+        {
+            var info = _bindings.Infos[iinfo];
+            var outInfo = _bindings.OutputCols[iinfo];
+            var tfType = _bindings.OutputTFTypes[iinfo];
+
+            if (tfType == TFDataType.Float)
+            {
+                return Utils.MarshalInvoke(MakeGetter<float>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            if (tfType == TFDataType.Double)
+            {
+                return Utils.MarshalInvoke(MakeGetter<double>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            if (tfType == TFDataType.Int32)
+            {
+                return Utils.MarshalInvoke(MakeGetter<int>, outInfo.ItemType.RawType, input, iinfo);
+            }
+
+            if (tfType == TFDataType.Int64)
+            {
+                return Utils.MarshalInvoke(MakeGetter<Int64>, outInfo.ItemType.RawType, input, iinfo);
```

Comment:
>Int64 [](start = 54, length = 5)

Do you have any tests testing this (and int, bool and string)?

Reply:
Not yet because ML.Net does not support dotnet types it uses DvInt etc. Once ZeeshanS changes are there I will be able to test.

---
In reply to: [210684078](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210684078) [](ancestors = 210684078)

Reply:
I don't know when Zeeshan's changes will be done, but for now I don't think this would work. If you don't want to handle these types using DvInt,DvBool etc., then I think we should not support them until the type system changes go in.

---
In reply to: [210718189](https://github.com/zeahmed/machinelearning/pull/3#discussion_r210718189) [](ancestors = 210718189,210684078)